gui/main.py

```diff
@@ -0,0 +1,248 @@
+        self.lowerRangeLabel.setGeometry(QRect(70, 120, 121, 31))
+        self.lowerRangeLabel.setFont(font)
+
+        self.higherRangeLabel = QLabel("Higher Range", self)
+        self.higherRangeLabel.setGeometry(QRect(340, 120, 121, 31))
+        self.higherRangeLabel.setFont(font)
+
+        self.portRangeLabel = QLabel("Port Range", self)
+        self.portRangeLabel.setGeometry(QRect(50, 80, 121, 31))
+        self.portRangeLabel.setFont(font)
+
+        self.dividerLine = QFrame(self)
+        self.dividerLine.setGeometry(QRect(175, 90, 391, 16))
+        self.dividerLine.setFrameShape(QFrame.HLine)
+        self.dividerLine.setFrameShadow(QFrame.Sunken)
+
+        self.lowRangeSpinner = QSpinBox(self)
+        self.lowRangeSpinner.setMaximum(65535)
+        self.lowRangeSpinner.setGeometry(QRect(180, 120, 91, 31))
+
+        self.highRangeSpinner = QSpinBox(self)
+        self.highRangeSpinner.setMaximum(65535)
+        self.highRangeSpinner.setGeometry(QRect(460, 120, 91, 31))
+
+        self.createProgressBar()
+
+        self.scanResultsLabel = QLabel("Scan Results",self)
+        self.scanResultsLabel.setGeometry(QRect(270, 250, 121, 31))
+        self.scanResultsLabel.setFont(font)
+
+        self.resultTextField = QPlainTextEdit(self)
+        self.resultTextField.setGeometry(QRect(35, 290, 570, 150))
+        self.resultTextField.setReadOnly(True)
+
+        self.statusBar().showMessage('Ready')
+        self.statusBar().setStyleSheet("background-color : grey")
+
+
+        self.createMenuBar()
+
+        
+        self.setFixedSize(640, 480)
+        self.show()
+
+    def createProgressBar(self):
+
+        self.scanProgressBar = QProgressBar(self)
+        self.scanProgressBar.setGeometry(QRect(130, 220, 401, 23))
+        self.scanProgressBar.setHidden(False)
+        self.scanProgressBar.setValue(0)
+        #self.startProgressBar()
+
+    def startProgressBar(self):
+        self.threadClass = self.ThreadClass()
+        self.threadClass.finished.connect(self.scanCompleted)
+        self.threadClass.start()
+        self.scanProgressBar.setValue(0)
+    
+
+        lowRange = int(self.lowRangeSpinner.value())
+        highRange = int(self.highRangeSpinner.value())
+
+
+        hostAddress = str(self.hostText.toPlainText())
+        self.threadClass.remoteServer = hostAddress
+
+        self.threadClass.lowRange = lowRange
+        self.threadClass.highRange = highRange
+
+        self.threadClass.cng_val.connect(self.updateProgressBar)
+        self.threadClass.log_message.connect(self.newLog)
+
+    def newLog(self, log): 
+        self.resultTextField.insertPlainText(log)
+
+    def createMenuBar(self):
+        exitAct = QAction(QIcon('exit.png'), '&Exit', self)
+        exitAct.setShortcut('Ctrl+Q')
+        exitAct.setStatusTip('Exit application')
+        exitAct.triggered.connect(qApp.quit)
+
+        saveAct = QAction(QIcon('save.png'), '&Save Config', self)
```

Comment:
@Sohaib03 Where is `save.png`?

Reply:
I am thinking not to add icons. Since adding icon will reduce cross platform native look.

Reply:
I have added native icons from QTStyle. I hope they work on MAC.

Reply:
Alright!